packages/otelbin/src/components/validation-type/BackendValidation.tsx

```diff
@@ -4,31 +4,34 @@
 import { Accordion, AccordionContent, AccordionItem, AccordionTrigger } from "~/components/accordion";
 import { Select, SelectContent, SelectGroup, SelectItem, SelectTrigger, SelectValue } from "~/components/select";
 import { Tooltip, TooltipContent, TooltipTrigger } from "~/components/tooltip";
-import { useDistributions } from "../validation/useDistributions";
 import OtelLogo from "./../assets/svg/otel.svg";
 import { Github, Globe } from "lucide-react";
 import { Button } from "../button";
+import { preload } from "swr";
 import { IconButton } from "../icon-button";
 import { CurrentBadge } from "./ValidationTypeContent";
 import type { ICurrentValidation } from "./ValidationType";
-import { useState } from "react";
+import { useUrlState } from "~/lib/urlState/client/useUrlState";
+import { useRouter } from "next/navigation";
+import { distroBinding, distroVersionBinding } from "../validation/binding";
+import type { Distributions } from "~/types";
+
+const fetcher = (url: string) => fetch(url).then((res) => res.json());
+
+preload(`https://www.otelbin.io/validation/supported-distributions`, fetcher);
 
 export default function BackendValidation({
-	current,
-	setCurrent,
+	currentDistro,
+	setCurrentDistro,
+	data,
 }: {
-	current: ICurrentValidation;
-	setCurrent: (current: ICurrentValidation) => void;
+	currentDistro: ICurrentValidation;
+	setCurrentDistro: (current: ICurrentValidation) => void;
```

Comment:
**Question:** Why do we need `setCurrentDistro` here? We already have `getUrl` accessible within the file that we can use to change the distro.